enterprise/models.py

```diff
@@ -807,6 +830,39 @@ def consent_needed(self):
         enterprise_customer = self.enterprise_customer_user.enterprise_customer
         return enterprise_customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_ENROLLMENT)
 
+    @property
+    def audit_reporting_disabled(self):
+        """
+        Specify whether audit track data reporting is disabled for this enrollment.
+
+        * If the enterprise customer associated with this enrollment enables audit track data reporting,
+          simply return False.
+        * If the enterprise customer associated with this enrollment does not enable audit track data reporting,
+          return True if we are dealing with an audit enrollment, and False otherwise.
+
+        :return: True if audit track data reporting is disabled, False otherwise.
+        """
+        if not self.enterprise_customer_user.enterprise_customer.enables_audit_data_reporting:
+            return self.is_audit_enrollment
+
+        # Since audit data reporting is enabled, we always return False here.
+        return False
+
+    @property
+    def is_audit_enrollment(self):
+        """
+        Specify whether the course enrollment associated with this ``EnterpriseCourseEnrollment`` is in audit mode.
+
+        :return: Whether the course enrollment mode is of an audit type.
+        """
+        course_enrollment_api = EnrollmentApiClient()
+        course_enrollment = course_enrollment_api.get_course_enrollment(
+            self.enterprise_customer_user.username,
+            self.course_id
+        )
+        audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit', 'honor'])
+        return course_enrollment.get('mode') in audit_modes
```

Comment:
I ran into the following error when testing, so I would suggest changing this to:
`return course_enrollment and core_enrollment.get('mode') in audit_modes`

But, this raises an interesting question about how this scenario might come about. I'm pretty sure what happened is that the user unenrolled from the course, so the record in student_courseenrollment was removed, but the enterprise_enterprisecourseenrollment record still exists (which is what I'm seeing in the database). I think the right thing to do is to not send any data back in this case, since the course can't be completed if the user unenrolled. We'll have to consider what to report back to indicate that they unenrolled, but that is for another time.

To move forward with testing, I deleted the two enterprise enrollment records. We can just recreate this case by going through the flow and then unenrolling.

```
2017-08-03 10:36:27,690 ERROR 4962 [enterprise.api_client.lms] lms.py:223 - no course enrollment details found for username=mtd_learner8, course=course-v1:SAPSuccessFactors+SAPSF101+2017_R1
Traceback (most recent call last):
  File "./manage.py", line 116, in <module>
    execute_from_command_line([sys.argv[0]] + django_args)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/transmit_learner_data.py", line 50, in handle
    self.transmit_learner_data.delay(api_username, integrated_channel.channel_code(), integrated_channel.pk)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/__init__.py", line 24, in no_delay
    func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/transmit_learner_data.py", line 60, in transmit_learner_data
    integrated_channel.transmit_learner_data(api_user)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/models.py", line 101, in transmit_learner_data
    for learner_data in exporter.collect_learner_data():
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/learner_data.py", line 118, in collect_learner_data
    if self._should_skip_enrollment(enterprise_enrollment):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/learner_data.py", line 150, in _should_skip_enrollment
    enterprise_enrollment.audit_reporting_disabled,
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/models.py", line 855, in audit_reporting_disabled
    return self.is_audit_enrollment
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/models.py", line 873, in is_audit_enrollment
    return course_enrollment.get('mode') in audit_modes
AttributeError: 'NoneType' object has no attribute 'get'
```